pkg/network/common/egress_dns.go

```diff
@@ -96,13 +123,13 @@ func (e *EgressDNS) Sync() {
 				// Item needs to wait for this duration before it can be processed
 				duration = tm.Sub(now)
 			} else {
-				err, changed := e.Update(policyUID)
+				err, changed := e.Update(dnsName)
 				if err != nil {
 					utilruntime.HandleError(err)
 				}
 
 				if changed {
-					e.Updates <- EgressDNSUpdate{policyUID, policyNamespace}
+					e.Updates <- EgressDNSUpdate{updates}
```

Comment:
so rather than wrapping the list of updates, you could just do

    for _, update := range updates {
            e.Updates <- update
    }

and then you don't need to change `egress_network_policy.go` or `proxy.go` at all

Reply:
I would rather we batch all the updates as a single channel update to the `Sync` observer. That way we don't incur un-necessary overhead of tic-toc'ing between the two go-routines.

Reply:
If you want to do that then I'd say just return the array of changes directly rather than wrapping it up in a struct